pkg/otel/common/schema/builder/duration.go

```diff
@@ -54,10 +54,11 @@ func (b *DurationBuilder) Append(value arrow.Duration) {
 		return
 	}
 
-	if value != 0 {
+	if value != 0 && b.updateRequest != nil {
 		// If the builder is nil, then the transform node is not optional.
 		b.transformNode.RemoveOptional()
-		b.updateRequest.Inc()
+		b.updateRequest.Inc(&update.NewFieldEvent{FieldName: b.transformNode.Path()})
+		b.updateRequest = nil // No need to report this again.
```

Comment:
For the pattern here -- should the `b.updateRequest != nil` condition apply to lines 60-61 and not 59?
I.e., how come this is not
```
if value != 0 {
		// If the builder is nil, then the transform node is not optional.
		b.transformNode.RemoveOptional()
        if b.updateRequest != nil 
		        b.updateRequest.Inc(&update.NewFieldEvent{FieldName: b.transformNode.Path()})
		        b.updateRequest = nil 
       }
}
```

Reply:
It's intentional. In fact the previous version was not optimal and was doing the same idempotent node transformation again and again for each value added to the builder. This modification has the same effect on the tree of transformations while doing less modification on it.